File: web-automation/src/test/java/com/cholemetric/web/pages/DashboardPage.java
package com.cholemetric.web.pages;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import java.util.List;
import org.openqa.selenium.WebElement;

public class DashboardPage extends BasePage {
    private static final By NEW_ANALYSIS_BTN = By.cssSelector(".btn-new-analysis, #btnNewAnalysis, a[href*='new_analysis']");
    private static final By PATIENT_HISTORY_BTN = By.cssSelector(".btn-history, #btnPatientHistory, a[href*='patient_history']");
    private static final By SETTINGS_BTN = By.cssSelector(".btn-settings, a[href*='settings']");
    private static final By LOGOUT_BTN = By.cssSelector(".btn-logout, #btnLogout, button.logout");
    private static final By WELCOME_MESSAGE = By.cssSelector(".welcome-msg, .doctor-name, h1, h2");
    private static final By SCAN_COUNT = By.cssSelector(".scan-count, .total-scans, #scanCount");
    private static final By RECENT_SCANS = By.cssSelector(".recent-scans, .scan-list, #recentScans");
    private static final By EMPTY_STATE = By.cssSelector(".empty-state, .no-scans");

    public DashboardPage(WebDriver driver) { super(driver); }

    public void navigateTo(String baseUrl) { driver.get(baseUrl + "dashboard.html"); }
    public void clickNewAnalysis() { click(NEW_ANALYSIS_BTN); }
    public void clickPatientHistory() { click(PATIENT_HISTORY_BTN); }
    public void clickSettings() { click(SETTINGS_BTN); }
    public void clickLogout() { click(LOGOUT_BTN); }
    public boolean isDashboardDisplayed() { return driver.getTitle().contains("Dashboard") || isDisplayed(NEW_ANALYSIS_BTN) || isDisplayed(WELCOME_MESSAGE); }
    public boolean isWelcomeMessageDisplayed() { return isDisplayed(WELCOME_MESSAGE); }
    public boolean isNewAnalysisBtnDisplayed() { return isDisplayed(NEW_ANALYSIS_BTN); }
    public boolean isLogoutBtnDisplayed() { return isDisplayed(LOGOUT_BTN); }
    public String getWelcomeText() { try { return getText(WELCOME_MESSAGE); } catch(Exception e) { return ""; } }
    public List<WebElement> getRecentScans() { return driver.findElements(RECENT_SCANS); }
    public boolean isEmptyStateDisplayed() { return isDisplayed(EMPTY_STATE); }
}

File: automation/selenium_tests/pages/dashboard_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage

class DashboardPage(BasePage):
    """Page Object Model for Dashboard Page."""
    
    # Locators
    DOCTOR_NAME_HEADER = (By.ID, "display_doctor_name")
    NEW_ANALYSIS_BTN = (By.XPATH, "//a[contains(@href, 'new_analysis')]")
    PATIENT_HISTORY_BTN = (By.XPATH, "//a[contains(@href, 'patient_history')]")
    SETTINGS_BTN = (By.XPATH, "//a[contains(@href, 'settings')]")
    LOGOUT_BTN = (By.ID, "btn_logout")
    TOTAL_SCANS_CARD = (By.ID, "stat_total_scans")
    CONFIRMED_STONES_CARD = (By.ID, "stat_positive_cases")

    def navigate_to_dashboard(self, base_url):
        self.open(f"{base_url}/dashboard.html")

    def is_dashboard_loaded(self):
        return self.is_displayed(self.NEW_ANALYSIS_BTN) or self.is_displayed(self.PATIENT_HISTORY_BTN)

    def click_new_analysis(self):
        self.click(self.NEW_ANALYSIS_BTN)

    def click_patient_history(self):
        self.click(self.PATIENT_HISTORY_BTN)
